src/to_square_example.c

```diff
@@ -0,0 +1,177 @@
+  gdraw_command_image_draw(ctx, temp_copy, GPoint(0, 0));
+  free(temp_copy);
+}
+
+//Animation update callback
+
+static void update_icon_square_normalized(Animation *animation, const uint32_t distance_normalized) {
+  to_square_normalized = distance_normalized;
+  layer_mark_dirty(icon_layer);
+}
+
+static const PropertyAnimationImplementation s_icon_square_normalized_implementation = {
+  .base = {
+    .update = (AnimationUpdateImplementation) update_icon_square_normalized,
+  },
+};
+
+//function prototype
+static void stopped_animation_handler(Animation *animation, bool finished, void *context);
+
+static void schedule_animation_to_square(){
+  Animation * animation_to_square = (Animation *) property_animation_create(&s_icon_square_normalized_implementation, images[0], NULL, NULL);
+  animation_set_duration(animation_to_square, 1200);
+  animation_set_curve(animation_to_square, AnimationCurveEaseInOut);
+  animation_set_handlers(animation_to_square, (AnimationHandlers) {
+    .stopped = stopped_animation_handler,
+  }, NULL);
+
+
+
+
+  animation_schedule(animation_to_square);
+}
+
+static void schedule_animation_from_square(){
+
+  Animation *animation_from_square = (Animation *) property_animation_create(&s_icon_square_normalized_implementation, images[1], NULL, NULL);
+  animation_set_duration(animation_from_square, 1200);
+  animation_set_reverse(animation_from_square, true);
+  animation_set_handlers(animation_from_square, (AnimationHandlers) {
+    .stopped = stopped_animation_handler,
+  }, NULL);
+
+
+
+  animation_schedule(animation_from_square);
+}
+
+//Animation stopped callback handler. Switches between the animations
+
+static void stopped_animation_handler(Animation *animation, bool finished, void *context) {
+    current_image = (current_image+1)%2;
+    switch(current_image){
+      case 0: schedule_animation_to_square();break;
+      default: schedule_animation_from_square();break;
+    }
+  }
+
+
+
+static void main_window_load(Window *window) {
+  Layer *window_layer = window_get_root_layer(window);
+  layer_set_update_proc(window_layer, bg_layer_update_proc);
+  GRect icon_rect = GRect(50, 50, 50, 50);
+  icon_layer = layer_create(icon_rect);
+  layer_set_update_proc(icon_layer, icon_layer_update_proc);
+  layer_add_child(window_layer, icon_layer);
+
+  schedule_animation_to_square();
+}
+
+static void main_window_unload(Window *window) {
+  layer_destroy(icon_layer);
+}
+
+
+static void init() {
+  images[0] = gdraw_command_image_create_with_resource(RESOURCE_ID_ICON_HEAVY_RAIN);
+  images[1] = gdraw_command_image_create_with_resource(RESOURCE_ID_ICON_DOCUMENT);
+  current_image = 0;
+  to_square_normalized = 0;
+
+  s_main_window = window_create();
+  window_set_window_handlers(s_main_window, (WindowHandlers) {
+    .load = main_window_load,
+    .unload = main_window_unload,
+  });
+  window_stack_push(s_main_window, true);
+}
+
+static void deinit() {
+  for(int j = 0; j < IMAGES_LENGTH; j++)free(images[j]);
```

Comment:
Again, do not directly `free()` SDK objects. Use the corresponding `_destroy()` function instead.